install/kubernetes/helm/istio/files/injection-template.yaml

```diff
@@ -1,9 +1,11 @@
 rewriteAppHTTPProbe: {{ valueOrDefault .Values.sidecarInjectorWebhook.rewriteAppHTTPProbe false }}
-{{- if or (not .Values.istio_cni.enabled) (annotation .ObjectMeta `sidecar.istio.io/enableCoreDump` .Values.global.proxy.enableCoreDump) }}
 initContainers:
 {{ if ne (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `NONE` }}
-{{- if not .Values.istio_cni.enabled }}
+{{ if .Values.istio_cni.enabled -}}
+- name: istio-validation
```

Comment:
For back compatibility: 
There was a tricky logic: istio-cni doesn't set up iptables if there is an existing istio-init container to avoid an non-cni istio-init container and istio-cni manipulate the iptables independently.

I don't have the idea behind it. My guess is to support migrating to/from istio-cni

Reply:
Ah, fair enough. That works for me